lib/posix/options/Kconfig.mem

```diff
@@ -49,11 +51,13 @@ config POSIX_MEMLOCK
 
 config POSIX_MEMLOCK_RANGE
 	bool "POSIX range memory locking"
-	imply MMU if (CPU_HAS_MMU && ARCH_HAS_DEMAND_PAGING)
-	imply DEMAND_PAGING
```

Comment:
I think `imply DEMAND_PAGING` will need to remain unless it is defaulted to `y` with `ARCH_HAS_DEMAND_PAGING`